test/e2e/cross_os/test_job_submissions.py

```diff
@@ -312,6 +313,498 @@ def is_expected_session_action_canceled(sessions: List[Dict[str, Any]]) -> bool:
+        job_details = Job.get_job_details(
+            client=deadline_client,
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            job_id=job_id,
+        )
+        job = Job(
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            template={},
+            **job_details,
+        )
+
+        @backoff.on_predicate(
+            wait_gen=backoff.constant,
+            max_time=60,
+            interval=2,
+        )
+        def sync_input_action_started(current_job: Job) -> bool:
+            sessions = deadline_client.list_sessions(
+                farmId=current_job.farm.id, queueId=current_job.queue.id, jobId=current_job.id
+            ).get("sessions")
+            if len(sessions) == 0:
+                return False
+            for session in sessions:
+                session_actions = deadline_client.list_session_actions(
+                    farmId=job.farm.id,
+                    queueId=job.queue.id,
+                    jobId=job.id,
+                    sessionId=session["sessionId"],
+                ).get("sessionActions")
+                logging.info(f"Session actions: {session_actions}")
+                for session_action in session_actions:
+                    if "syncInputJobAttachments" in session_action["definition"]:
+                        if session_action["status"] in ["ASSIGNED", "RUNNING"]:
+                            return True
+            return False
+
+        # Wait until the sync input action has started
+        assert sync_input_action_started(job)
+
+        deadline_client.update_job(
+            farmId=job.farm.id,
+            queueId=job.queue.id,
+            jobId=job.id,
+            targetTaskRunStatus="CANCELED",
+        )
+
+        @backoff.on_predicate(
+            wait_gen=backoff.constant,
+            max_time=120,
+            interval=10,
+        )
+        def sync_input_actions_are_canceled(sessions: List[Dict[str, Any]]) -> bool:
+            found_canceled_sync_input_action = False
+            for session in sessions:
+                session_actions = deadline_client.list_session_actions(
+                    farmId=job.farm.id,
+                    queueId=job.queue.id,
+                    jobId=job.id,
+                    sessionId=session["sessionId"],
+                ).get("sessionActions")
+                logging.info(f"Session actions: {session_actions}")
+                for session_action in session_actions:
+                    # Session action should be canceled if it's the action we expect to be canceled
+                    if "syncInputJobAttachments" in session_action["definition"]:
+                        if session_action["status"] == "CANCELED":
+                            found_canceled_sync_input_action = True
+                    else:
+                        assert (
+                            session_action["status"] != "CANCELED"
```

Comment:
Given the comment below, should the assertion here then be:

```
session_action["status"] == "SUCCEEDED" or session_action["status"] == "NEVER_ATTEMPTED" 
```

Reply:
I guess the issue here is that I'm not waiting for the job to be completed, which means that other actions could be in some other state, such as RUNNING.

I think it makes sense that I wait for the job to be completed